sandbox/pages/StatusInputPage.qml

```diff
@@ -146,7 +146,7 @@ Column {
     StatusInput {
         label: "Label"
         input.placeholderText: "Input width component (right side)"
-        input.component: StatusIcon {
+        input.rightComponent: StatusIcon {
```

Comment:
This is a breaking change. We need to be very careful with merging this.
I'd rather ensure `component` is still working and introduce a `rightComponent` property that acts as an alias.

Reply:
Yes, that's right, it's a breaking change. Thanks for pointing that out. The status app needs to adapt.

Current StatusInput interface is:
```
property Item leftComponent
property Item component
```
it's inconsistent, `component` says nothing about its position, how could clients possibly interpret that? How will they know that `component` is what is meant to be `rightComponent`?

I could make an alias for that to not break clients' code. How long are we going to maintain this misleading property? Do we have any process for deprecation in place? Cause for me it would be a candidate for deprecation for sure.

Reply:
> it's inconsistent, component says nothing about its position, how could clients possibly interpret that? How will they know that component is what is meant to be rightComponent?

This is because `component` was introduced way before we knew `leftComponent` will be needed. So originally there was only `component`. One of the common things when software historically grows, even more important that we avoid breaking consumers (which is why I haven't changed `component` back then).

> I could make an alias for that to not break clients' code. How long are we going to maintain this misleading property? Do we have any process for deprecation in place? Cause for me it would be a candidate for deprecation for sure.

We don't have an official deprecation process in place, partly also because we're still in 0.x so theoretically, breaking changes are allowed without further notice. I'd still like to keep them at a minimum/push them back because StatusQ is now heavily used inside Desktop and it'd be bad if we'd break it.

I agree it's good to think about a deprecation process. Could imagine, once a thing is marked as deprecated it can be removed in the next major version, giving consumers time to adapt. 

But again, probably only makes sense once we've reached 1.x.

I don't think at least, that introducing a property alias here is going to hurt maintenance that much.

Reply:
> I don't think at least, that introducing a property alias here is going to hurt maintenance that much.

It doesn't hurt maintenance that much, it hurts the interface readability.

```
StatusInput {
  component: WhatToPutHere{}
}
```
If you were the client, what would the `component` mean to you? What would you need to do to figure that out?

> I'd still like to keep them at a minimum/push them back because StatusQ is now heavily used inside Desktop and it'd be bad if we'd break it.

At some point, we will need to remove these deprecated components/properties anyway. In my opinion, it's better to do it now, since the number of clients can only grow in time (and so the number of changes required to adapt).

> We don't have an official deprecation process in place, partly also because we're still in 0.x so theoretically, breaking changes are allowed without further notice. I'd still like to keep them at a minimum/push them back because StatusQ is now heavily used inside Desktop and it'd be bad if we'd break it.

If we have that possibility right now, we should use it as much as possible (when something is clearly wrong).

Reply:
> If you were the client, what would the component mean to you? What would you need to do to figure that out?

Ideally this is solved by documentation. Naming is hard.

Reply:
If I've not missed something, I've made a quick find in the desktop project and I think it will be easy to update if we remove the `component` property so.. if it does not require so much effort, we can remove it and leave the component clean.